src/elm/Pages/Build/Logs.elm

```diff
@@ -342,27 +349,91 @@ downloadFileName org repo buildNumber resourceType resourceNumber =
     String.join "-" [ org, repo, buildNumber, resourceType, resourceNumber ]
 
 
+{-| processLog : takes a log as string, splits it by newline, and processes it into a model that can render custom elements like timestamps and links
+-}
+processLog : String -> List (LogLine msg)
+processLog log =
+    log
+        |> String.split "\n"
+        |> List.map
+            (\log_ ->
+                let
+                    -- first we convert the log into an ANSI Log Model
+                    -- this lets us preserve the original ANSI style while applying our own processing
+                    ansiLogModel =
+                        Ansi.Log.update log_ defaultANSILogModel
+
+                    -- next we take the ANSI Model, run custom processing like link parsing, then render it to Html
+                    logView =
+                        -- we need to ignore extra empty lines generated by Ansi.Log.update
+                        case List.head <| Array.toList ansiLogModel.lines of
+                            Just logLine ->
+                                -- we may need to extend this to accept user preferences
+                                -- for example, a user may choose to not render links
+                                processAndViewLine logLine
+
+                            Nothing ->
+                                text ""
+                in
+                -- pack the log into a model to make it more flexible when rendering later
+                -- this is particularly useful for adding toggleable features like timestamps
+                LogLine logView
+            )
+
+
+{-| parseLinksInChunk : takes Ansi.Log.Chunk and performs additional processing to parse links
+-}
+parseLinksInChunk : Ansi.Log.Chunk -> List (Html msg)
+parseLinksInChunk chunk =
+    let
+        -- split the "line" by escape characters
+        -- its possible this will split a "valid" link containing quote characters, but its a willing risk
+        split =
+            List.foldl Util.splitIntersperseConcat [ chunk.text ] linkEscapeCharacters
+    in
+    -- "process" each "split chunk" and check for link
+    List.map
+        (\chunk_ ->
+            case Url.fromString chunk_ of
+                Just link ->
+                    -- use toString in href to make the link safe
+                    a [ Util.testAttribute "log-line-link", href <| Url.toString link ] [ text chunk_ ]
+
+                Nothing ->
+                    text chunk_
+        )
+        split
+
+
+{-| linkEscapeCharacters : list of string escape characters that delimit links.
+for example "<https://github.com"> should be split from the quotes, even though " is a valid URL character (see: see: <https://www.rfc-editor.org/rfc/rfc3986#section-2>)
+-}
+linkEscapeCharacters : List String
+linkEscapeCharacters =
+    [ "'", " ", "\"", "\t", "\n" ]
```

Comment:
there might be more values to add here, i nearly forgot `\t` and caught it during testing.